go/merkletree_new/merkle_tree_interface.go

```diff
@@ -0,0 +1,65 @@
+package merkletree
+
+// MerkleTreeInterface is the methods which are required to manipulate the
+// contents of a Merkle Hash Tree (that is, add new entries), and to query
+// the tree for inclusion and transparency proofs.
+type MerkleTreeInterface interface {
+	// CurrentRootHash returns the hash of the entire current tree.
+	CurrentRootHash func () ([]byte, error)
```

Comment:
Yep, I can certainly see the value in having a separate `Hash` type.  The syntax issues are just my inexperience with Go shining through.